vllm/v1/engine/llm_engine.py

```diff
@@ -99,10 +98,21 @@ def from_engine_args(
 
     @classmethod
     def _get_executor_cls(cls, vllm_config: VllmConfig):
-        return GPUExecutor
+        distributed_executor_backend = (
+            vllm_config.parallel_config.distributed_executor_backend)
+        if distributed_executor_backend == "mp":
+            from vllm.v1.executor.multiproc_gpu_executor import (
+                MultiprocessingGPUExecutor)
+            executor_class = MultiprocessingGPUExecutor
+        else:
+            assert (distributed_executor_backend is None)
+            from vllm.v1.executor.gpu_executor import GPUExecutor
+            executor_class = GPUExecutor
+
+        return executor_class
 
     def stop_remote_worker_execution_loop(self) -> None:
```

Comment:
QQ: what is this API for? IIUC, we don't need this API anymore with new scheduler <> worker decoupled design?

Reply:
Are you talking about `stop_remote_worker_execution_loop` or `_get_executor_cls`?

For `stop_remote_worker_execution_loop` I am not sure -- honestly I stumbled across this while I was trying to debug some process termination bugs.

Reply:
@rkooo567 yes you're right, this will be no longer needed/used since we'll have a single event loop in the workers, we should remove it!

Reply:
I'll take it out 👍